src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -411,26 +418,36 @@ public static function convertWooCommerceToStorekeeperOrderStatus($wc_order_stat
      * @param WC_Order $databaseOrder - Order items to compare from
      * @param $backofficeOrder - Order items to compare to
      */
-    protected function checkOrderDiscrepancy(WC_Order $databaseOrder, $backofficeOrder): bool
+    protected function checkOrderDifference(WC_Order $databaseOrder, $backofficeOrder): bool
     {
-        $hasDiscrepancy = false;
+        $hasDifference = false;
         $databaseOrderItems = $this->getOrderItems($databaseOrder);
         $backofficeOrderItems = $backofficeOrder['order_items'];
 
         foreach ($databaseOrderItems as $databaseOrderItem) {
             $sku = $databaseOrderItem['sku'];
             if (!in_array($sku, array_column($backofficeOrderItems, 'sku'), true)) {
-                $hasDiscrepancy = true;
+                $hasDifference = true;
+                break;
+            }
+
+            $index = array_search($sku, array_column($backofficeOrderItems, 'sku'), true);
+            $backofficeOrderItem = $backofficeOrderItems[$index];
+            if (
+                $backofficeOrderItem['quantity'] !== $databaseOrderItem['quantity'] ||
+                $backofficeOrderItem['ppu_wt'] !== $databaseOrderItem['ppu_wt']
+            ) {
+                $hasDifference = true;
                 break;
```

Comment:
You really need to make 2 [sets](https://en.wikipedia.org/wiki/Set_(abstract_data_type)) and compare if those are the same. This one searched for the first product, so if there are 2 with the same sku it wont work

Sets like this will do just good: `Set( {sku,quantity,ppu_wt}, ... ) `

Reply:
I think it's good to add unit test just for the comparison part, cos there can be many cases there

Reply:
Added checking by set, and unit tests.